performance-timeline/po-entries-sort.html

```diff
@@ -0,0 +1,101 @@
+<!DOCTYPE HTML>
+<head>
+<meta charset="utf-8">
+<title>PerformanceObservers: getEntries* sort order</title>
+<meta name="author" title="JosephPecoraro" href="mailto:joepeck@webkit.org">
+<meta name="assert" content="PerformanceObserverEntryList entries are sorted correctly">
+<link rel="help" href="https://w3c.github.io/performance-timeline/#filter-performance-entry-buffer-by-name-and-type">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="performanceobservers.js"></script>
+</head>
+<h1>PerformanceObservers: getEntries* sort order</h1>
+<p>getEntries, getEntriesByType and getEntriesByName sort order</p>
+<div id="log"></div>
+<script>
+  async_test(function (t) {
+    var stored_entries = [];
+    var stored_entries_by_type = [];
+    var observer = new PerformanceObserver(
+        t.step_func(function (entryList, obs) {
+
+          stored_entries = entryList.getEntries();
+          stored_entries_by_type = entryList.getEntriesByType("mark");
+          stored_entries_by_name = entryList.getEntriesByName("name-repeat");
+          var startTimeOfMark2 = entryList.getEntriesByName("mark2")[0].startTime;
+
+          checkSorted(stored_entries);
+          checkEntries(stored_entries, [
+            {entryType: "measure", name: "measure1"},
+            {entryType: "measure", name: "measure2"},
+            {entryType: "measure", name: "measure3"},
+            {entryType: "measure", name: "name-repeat"},
+            {entryType: "mark", name: "mark1"},
+            {entryType: "mark", name: "mark2"},
+            {entryType: "measure", name: "measure-matching-mark2-1"},
+            {entryType: "measure", name: "measure-matching-mark2-2"},
+            {entryType: "mark", name: "name-repeat"},
+            {entryType: "mark", name: "name-repeat"},
+          ]);
+          checkEntriesInOrder(stored_entries, [
```

Comment:
Yes this seems like unnecessary complexity and the addition of the helper "checkEntriesInOrder" is misleading also.
checkSorted and checkEntries should suffice for everything that performance-observer should guarantee. Otherwise let's fix them instead of adding overlapping helper methods.